File: reto02-palindrome-number/java/PalindromeNumber.java
class PalindromeNumber {
  public static void main(String[] args) {
    System.out.println(isPalindromeNumber(121)); // true
    System.out.println(isPalindromeNumber(-121)); // false
    System.out.println(isPalindromeNumber(10)); // false
  }

  public static boolean isPalindromeNumber(int number) {
    if (number < 0) {
      return false;
    }

    int newNumber = number;
    int reversed = 0;
    while (newNumber > 0) {
      reversed = reversed * 10 + newNumber % 10;
      newNumber = newNumber / 10;
    }
    return number == reversed;
  }
}

File: reto02-palindrome-number/javascript/palindromeNumber.js
function main() {
  console.log(isPalindrome(121)); // true
  console.log(isPalindrome(-121)); // false
  console.log(isPalindrome(10)); // false
}

function isPalindrome(number) {
  if (number < 0) {
    return false;
  }

  let newNumber = number;
  let reversed = 0;
  while (newNumber > 0) {
    reversed = reversed * 10 + newNumber % 10;
    newNumber = parseInt(newNumber / 10);
  }
  return number == reversed;
}

main();
